src/Agent.java

```diff
@@ -18,8 +23,15 @@ public void setProperties(ArrayList<Property> properties) {
         this.properties = properties;
     }
     
-    public void addProperty(Property property) {
-        this.properties.add(property);
+    public void addProperty(Property property) throws InvalidOperationException{
+        if (property instanceof Property) {
+        	this.properties.add(property);	
+        }
+        else {
```

Comment:
```suggestion
        } else {
```
Format your code according to standards. I linked a link to a standards in our group channel. 
Take a look at all your files, i will not point other mistakes at this round of reviewing.